### PR TITLE
So 98 making splash screen v2

### PR DESCRIPTION
**Pull request by Tomman85**

### CODE REVIEW

File: lib/presentation/home_page/components/main_page_weather_content.dart [hunk 1]
```diff
@@ -132,10 +134,30 @@ class _MainPageWeatherContentState extends State<MainPageWeatherContent> {
...
+                        key: const ValueKey(0),
+                      ),
                     );
                   }
+                  return AnimatedSwitcher(
+                    // switchInCurve: Curves.fastOutSlowIn,
+                    // switchOutCurve: Curves.bounceIn,
+
+                    transitionBuilder: (child, animation) => FadeTransition(
+                      opacity: Tween<double>(begin: 0.0, end: 3.0).animate(
```
Comment (JulianGerman) [inline lib/presentation/home_page/components/main_page_weather_content.dart:149]: I think opacity needs only <0,1> values.

File: lib/presentation/home_page/components/main_page_weather_content.dart [hunk 2]
```diff
@@ -132,10 +134,30 @@ class _MainPageWeatherContentState extends State<MainPageWeatherContent> {
...
+                    child = Center(
+                      child: Lottie.asset(
+                        'lib/assets/lottie/61302-weather-icon.json',
+                        key: const ValueKey(0),
+                      ),
                     );
                   }
+                  return AnimatedSwitcher(
+                    // switchInCurve: Curves.fastOutSlowIn,
+                    // switchOutCurve: Curves.bounceIn,
```
Comment (JulianGerman) [inline lib/presentation/home_page/components/main_page_weather_content.dart:146]: Redundant comments and empty line.

File: lib/presentation/home_page/components/main_page_weather_content.dart [hunk 3]
```diff
@@ -69,12 +69,14 @@ class _MainPageWeatherContentState extends State<MainPageWeatherContent> {
...
                 ) {
+                  Widget child;
                   if (snapshot.hasData) {
                     dynamic data = snapshot.data;
                     double windConverter =
                         ((data.currentWeatherModel.windSpeed * 1 / 1000) /
                             (1 / 3600));
-                    return Container(
+                    child = Container(
+                      key: const ValueKey(1),
```
Comment (JulianGerman) [inline lib/presentation/home_page/components/main_page_weather_content.dart:79]: Try without that and without key in Lottie asset, these are different types so it should works without using keys.

File: lib/presentation/home_page/components/main_page_weather_content.dart [hunk 4]
```diff
@@ -132,10 +134,30 @@ class _MainPageWeatherContentState extends State<MainPageWeatherContent> {
...
+                    transitionBuilder: (child, animation) => FadeTransition(
+                      opacity: Tween<double>(begin: 0.0, end: 3.0).animate(
+                        CurvedAnimation(
+                          parent: animation,
+                          curve: Interval(0.7, 1.0),
+                        ),
+                      ),
+                      child: child,
+                    ),
+                    reverseDuration: const Duration(seconds: 18),
```
Comment (JulianGerman) [inline lib/presentation/home_page/components/main_page_weather_content.dart:157]: That 18 secs here looks a little bit weird here :) do we need that reverse duration at all?

**Review by JulianGerman — APPROVED** [review]
Let's try with these few changes, clean some code and feel free to merge it.👍🏼

File: lib/presentation/home_page/components/main_page_weather_content.dart (hunk 4 above)
Comment (Tomman85) [reply lib/presentation/home_page/components/main_page_weather_content.dart:157]: Yep we don't need reverse duration

File: lib/presentation/home_page/components/main_page_weather_content.dart (hunk 3 above)
Comment (Tomman85) [reply lib/presentation/home_page/components/main_page_weather_content.dart:79]: And we don't need this keys as well.  It works also without this keys

File: lib/presentation/home_page/components/main_page_weather_content.dart (hunk 1 above)
Comment (Tomman85) [reply lib/presentation/home_page/components/main_page_weather_content.dart:149]: Ok, so i set opacity in <0,1> values and it's works :)

**Pull request merged**